packages/runtime/src/next/api-handler/index.ts

```diff
@@ -131,8 +131,8 @@ export function MakeswiftApiHandler(
         )
     }
 
-    if ((m = matches<{ id: string }>('/page-pathname-slices/:id'))) {
-      return client.getPagePathnameSlice(m.params.id).then(handleResource)
+    if ((m = matches<{ id: string; locale?: string }>('/page-pathname-slices/:id/:locale?'))) {
```

Comment:
This is a confusing URL pattern because `locale` is not a sub resource of a page pathname slice. For REST APIs parameters like these are usually URL search parameters